code/modules/mob/living/carbon/carbon.dm

```diff
@@ -306,7 +306,25 @@
 							M.resting = 0
 
 				else if(istype(tapper))
-					tapper.species.tap(tapper,src)
+					if(on_fire)
+						tapper.species.tap(tapper,src)
+					else if(tapper.zone_sel.selecting in list(BP_L_HAND, BP_R_HAND, BP_L_ARM, BP_R_ARM))
+						var/emote_type = /singleton/overhead_emote/highfive
+						if(tapper.zone_sel.selecting == BP_L_HAND || tapper.zone_sel.selecting == BP_R_HAND)
+							emote_type = /singleton/overhead_emote/fistbump
+						else if(tapper.zone_sel.selecting == BP_L_ARM || tapper.zone_sel.selecting == BP_R_ARM)
+							emote_type = /singleton/overhead_emote/highfive
+
```

Comment:
not really a fan of how this is handled, should be a keyed list somewhere like on the species or whatever